cdap-ui/app/features/dashboard/services/chart-helpers.js

```diff
@@ -94,15 +96,15 @@ angular.module(PKG.name + '.feature.dashboard')
 
         // columns will be in the format: [ [metric1Name, v1, v2, v3, v4], [metric2Name, v1, v2, v3, v4], ... xCoords ]
         columns = [];
-        for (i = 0; i < newVal.length; i++) {
-          metricMap = newVal[i];
+        newVal.forEach(function (value) {
+          metricMap = value;
```

Comment:
Do we need to assign value to metricMap? it seems like we can just directly call Object.keys on value. 

Minor performance enhancement :)